src/cdk/a11y/_a11y.scss

```diff
@@ -18,29 +18,57 @@
   }
 }
 
-// Applies styles for users in high contrast mode. Note that this only applies
-// to Microsoft browsers. Chrome can be included by checking for the `html[hc]`
-// attribute, however Chrome handles high contrast differently.
-//
-// @param target Which kind of high contrast setting to target. Defaults to `active`, can be
-//    `white-on-black` or `black-on-white`.
-@mixin cdk-high-contrast($target: active) {
+/// Emits the mixin's content nested under `$selector-context` if `$selector-context`
+/// is non-empty.
+/// @param selector-context The selector under which to nest the mixin's content.
+@mixin _cdk-optionally-nest-content($selector-context) {
+  @if ($selector-context == '') {
+    @content;
+  } @else {
+    #{$selector-context} {
+      @content;
+    }
+  }
+}
+
+/// Applies styles for users in high contrast mode. Note that this only applies
+/// to Microsoft browsers. Chrome can be included by checking for the `html[hc]`
+/// attribute, however Chrome handles high contrast differently.
+///
+/// @param target Which kind of high contrast setting to target. Defaults to `active`, can be
+///    `white-on-black` or `black-on-white`.
+/// @param encapsulation Whether to emit styles for view encapsulation. Values are:
+///     * `on` - works for `Emulated`, `Native`, and `ShadowDom`
+///     * `off` - works for `None`
+///     * `both` - works for all encapsulation modes by emitting the CSS twice (default).
+@mixin cdk-high-contrast($target: active, $encapsulation: 'both') {
```

Comment:
I'm not sure the `both` value makes sense since encapsulation is either on or off.

Reply:
What would you do instead? I want the mixin to "just work" by default, even if it is emitting extra styles.

Reply:
I think the thing that's throwing me off is that it's called `both` which sounds like it can have more than one style encapsulations at the same time. Maybe can make it `on | off | undefined` and output both styles if it's emitted? Another approach could be to match the naming from Angular.

Reply:
I understand your point, but I think, `both` is fine here, because with this option, the mixin supports both general states of view/style encapsulation. So it's rather an option of which policies are supported.  
Matching the naming from Angular would be complicated, as AFAIK there's no explicit naming from Angular which addresses `Emulated`, `Native`, and `ShadowDom` together.

`undefined` is not a restricted value in Sass, so `null` would be the way to go. But I'm feeling that `both` fits better.

Reply:
How about `'any'`?

Reply:
Sure, that works too.

Reply:
Changed to `any`